src/map.rs

```diff
@@ -346,6 +346,52 @@ where
         self.put(key, value, false, guard)
     }
 
+    /// Removes all entries from this map.
+    pub fn clear<'g>(&'g self, guard: &'g Guard) {
+        // Negative number of deletions
+        let mut delta = 0;
+        let mut idx = 0usize;
+
+        let mut table = self.table.load(Ordering::SeqCst, guard);
+
+        // Safety: self.table is a valid pointer because we checked it above.
+        while !table.is_null() && idx < unsafe { table.deref() }.len() {
+            let tab = unsafe { table.deref() };
+            let node = tab.bin(idx, guard);
+            if node.is_null() {
+                idx = idx + 1;
+                continue;
+            }
+            // Safety: node is a valid pointer because we checked
+            // it in the above if stmt.
+            let node = unsafe { node.deref() };
+            match node {
+                BinEntry::Moved(next_table) => {
+                    table = self.help_transfer(table, *next_table, guard);
+                    idx = 0;
+                }
+                BinEntry::Node(ref node) => {
+                    let lock = node.lock.lock();
```

Comment:
We've called this `head_lock` in other parts of the code since you do not take the lock of _every_ node, just the first one. Let's continue to do that here :)